functions.py

```diff
@@ -1,84 +1,147 @@
+from command_factory import *
+from ICommand import ICommand
 import json
 import os
+from keys import *
 
 
-def search(path, primary_key):
-    y = False
-    for r, d, f in os.walk(path):
-        if primary_key in f:
-            y = True
-            break
-    return y
-
-
-def creates_dir(schema):
-    x = json.load(open(schema, 'r'))
-    p = os.path.join(os.getcwd(), x['database_name'])
-    os.mkdir(p)
-    for i in x['Tables']:
-        d = os.path.join(p, i['name'])
-        os.mkdir(d)
-
-
-def creates(schema, table, primary_key):
-    x = json.load(open(schema, 'r'))
-    path = os.getcwd() + '\\' + x['database_name'] + '\\' + table + '\\' + primary_key
-    f = open(path, 'w')
-    f.write("{\n")
-    i = 0
-    for k in x['Tables']:
-        if k['name'] != table:
-            i = i + 1
+
+
+class CreateDirCommand(ICommand):
+    def __init__(self, schema):
+        self.schema = schema
+
+    def execute(self):
+        if self.schema is None:
+            return "Schema not found"
+        data = json.load(open(self.schema, 'r'))
+        path = os.path.join(os.getcwd(), data[Keys().DATABASE])

```

Comment:
Move this to the constructor

Reply:
What is the idea about this, we made the constructor only gives the class variables their values, to be specific while every other function is run through the execute command.